torch/lib/c10d/Utils.cpp

```diff
@@ -231,6 +236,23 @@ int connect(
         if (!wait || (!anyRefused && !anyReset)) {
           throw;
         }
+
+        /*
+        if a timeout is specified, check time elapsed to see if we need to
+        timeout. A timeout is specified if timeout != kNoTimeout.
+        */
+        if (timeout != kNoTimeout) {
+          auto now =
+              std::chrono::high_resolution_clock::now().time_since_epoch();
+          auto nowMilliseconds =
+              std::chrono::duration_cast<std::chrono::milliseconds>(now)
+                  .count();
+
+          auto elapsedMilliseconds = nowMilliseconds - startMilliseconds;
+          if (elapsedMilliseconds > timeout.count()) {
+            throw std::runtime_error("Connecting to TCP store timed out.");
```

Comment:
There is a different timeout error thrown above, where `poll(2)` returns with a timeout. Please consolidate so there is only one message for connect timeout.

Reply:
Sounds good. The errors are slightly different though, one is because `poll(2)` timed out on a single attempt, while this error can happen because we keep getting `ECONNREFUSED` or `ECONNRESET` after trying many times. Is it worth having different error messages to help with debugging?

Reply:
I think it's fine to be the same and that multiple add to the confusion here.